src/draw/lv_image_decoder.c

```diff
@@ -126,11 +126,11 @@ lv_result_t lv_image_decoder_open(lv_image_decoder_dsc_t * dsc, const void * src
     /* Flush the D-Cache if enabled and the image was successfully opened */
     if(dsc->args.flush_cache && res == LV_RESULT_OK && dsc->decoded != NULL) {
         lv_draw_buf_flush_cache(dsc->decoded, NULL);
-        LV_LOG_INFO("Flushed D-cache: src %p (%s) (W%" LV_PRId32 " x H%" LV_PRId32 ", data: %p cf: %d)",
+        LV_LOG_INFO("Flushed D-cache: src %p (%s) (W%" LV_PRIu32 " x H%" LV_PRIu32 ", data: %p cf: %d)",
                     src,
                     dsc->src_type == LV_IMAGE_SRC_FILE ? (const char *)src : "c-array",
-                    dsc->decoded->header.w,
-                    dsc->decoded->header.h,
+                    (uint32_t)dsc->decoded->header.w,
```

Comment:
remove `(uint32_t)` cast

Reply:
From the error log and explanation above, we should remove the cast and change formatter to `%d`.

Reply:
Do you mean replace the original `LV_PRId32` with `%d`? Note the arguments really are `uint32_t` by definition, and the format strings for the exact variables in other locations use the unsigned spec `LV_PRIu32`.

Reply:
> Note the arguments really are uint32_t by definition

Yes. If I understand correctly, the type is converted to int unless it cannot fit in, and then converted to unsigned int.
> type ranks lower than int is automatically converted to the type int,

Let's see what others think about it.